hera_pspec/grouping.py

```diff
@@ -470,50 +548,56 @@ def average_spectra(uvp_in, blpair_groups=None, time_avg=False,
     uvp.Nblpairts = len(time_avg_arr)
     uvp.Nblpairs = len(np.unique(blpair_arr))
     uvp.Nbls = len(bl_arr)
-    
+
     # Baselines
     uvp.bl_array = bl_arr
     uvp.bl_vecs = bl_vecs
     uvp.blpair_array = np.array(blpair_arr)
-    
+
     # Times
     uvp.time_1_array = np.array(time_1)
     uvp.time_2_array = np.array(time_2)
     uvp.time_avg_array = np.array(time_avg_arr)
-    
+
     # LSTs
     uvp.lst_1_array = np.array(lst_1)
     uvp.lst_2_array = np.array(lst_2)
     uvp.lst_avg_array = np.array(lst_avg_arr)
-    
+
     # Data, weights, and no. samples
     uvp.data_array = data_array
     uvp.integration_array = ints_array
     uvp.wgt_array = wgts_array
     uvp.nsample_array = nsmp_array
+    if store_cov:
+        uvp.cov_array = cov_array
     if hasattr(uvp_in, 'label1'): uvp.label1 = uvp_in.label1
     if hasattr(uvp_in, 'label2'): uvp.label2 = uvp_in.label2
 
+    if error_field is not None:
```

Comment:
this logic of the following four lines doesn't make sense to me. can you double check this is indeed doing what you intend?

Reply:
This is the logic:

1. Check if input parameter "error" field is None. If it is not None, set attribute stats array to the object just constructed in the long for loops.
2. If error_field is None, but the uvp has an attribute called stats_array, delete it.  In other words, a uvp was given with a stats array but the user did not specify error bars to keep, so stats_array should not be in the output uvp at all.
3. If error_field is None _and_ stats_array is not an attribute of uvp, do nothing.

I hope this makes sense!

Reply:
Okay, that makes sense to me. thanks!

Reply:
except this can fail if `uvp` does not have a stats_array